modules/genomic_browser/php/endpoints/gwasbrowser.class.inc

```diff
@@ -0,0 +1,272 @@
+
+        return array(
+                'SNP_ID'       => $this->_formElementObjectCreator(
+                    'SNP_ID',
+                    'text',
+                    'SNP ID',
+                    null
+                ),
+                'Chromosome'   => $this->_formElementObjectCreator(
+                    'Chromosome',
+                    'text',
+                    'Chromosome',
+                    null
+                ),
+                'BP_Position'  => $this->_formElementObjectCreator(
+                    'BP_Position',
+                    'text',
+                    'BP Position',
+                    null
+                ),
+                'Major_Allele' => $this->_formElementObjectCreator(
+                    'Major_Allele',
+                    'select',
+                    'Major Allele',
+                    $alleleArray
+                ),
+                'Minor_Allele' => $this->_formElementObjectCreator(
+                    'Minor_Allele',
+                    'select',
+                    'Minor Allele',
+                    $alleleArray
+                ),
+                'MAF'          => $this->_formElementObjectCreator(
+                    'MAF',
+                    'text',
+                    'MAF',
+                    null
+                ),
+                'Estimate'     => $this->_formElementObjectCreator(
+                    'Estimate',
+                    'text',
+                    'Estimate',
+                    null
+                ),
+                'StdErr'       => $this->_formElementObjectCreator(
+                    'StdErr',
+                    'text',
+                    'Std Err',
+                    null
+                ),
+                'Pvalue'       => $this->_formElementObjectCreator(
+                    'Pvalue',
+                    'text',
+                    'P-value',
+                    null
+                ),
+               );
+    }
+
+    /**
+     * Permission verification.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     * @param array                  $values  The form values.
+     *
+     * @return array
+     */
+    private function _verifyPermissions(
+        ServerRequestInterface $request,
+        array $values
+    ) : array {
+        $user = $request->getAttribute('user');
+        if (!$user->hasPermission('genomic_browser_view_site')
+            || !$user->hasPermission('genomic_browser_view_allsites')
+        ) {
+            $values['error'] = 'Forbidden';
+            return $values;
+        }
+        return $values;
+    }
+
+    /**
+     * Used to create (select, text, ...) element for form filter.
+     *
+     * @param string     $name   for element
+     * @param string     $type   for element
+     * @param string     $label  for element
+     * @param array|null $values for element
+     *
+     * @return object
+     */
+    private function _formElementObjectCreator(
```

Comment:
Maybe this function can be moved to the parent class if it's needed by all the child classes.